src/read-models/equipment/get-all.ts

```diff
@@ -23,6 +24,12 @@ export const getAll = (
         O.map(area => ({
           ...equipment,
           areaName: area.name,
```

Comment:
At some point we probably should refactor this to stop repeatedly going over all events. As events grow this kind of shape will kill our performance.
For now lets now worry about.

Reply:
The current setup/architecture seems to revolve around repeatedly going through events - the fact this isn't particularly performant was my first ? when I looked through it.

I assume the idea behind the architecture is that:
- the simplicity in storing the data
- looking at the history
- adding new events
- benefits of using a read-only functional approach
- very easy testing

outweights the
1. additional complexity for simple operations like reading/updating the current state
2. performance impacts of repeatedly copying / making new arrays
3. performance impacts of looping through all events

with the potential mitigations of
1. using lots of helpers from RA / ones we write ourselves
3. index the events

2. is just a drawback of trying to do readonly stuff in javascript/typescript which, unlike a proper compiled language like rust, cannot make high performance optimisations